src/domain/users/users.repository.spec.ts

```diff
@@ -63,4 +70,78 @@ describe('UsersRepository', () => {
       expect(result).toEqual({ id: mockUser.id });
     });
   });
+
+  describe('addWalletToUser', () => {
+    it('should add a wallet to a user', async () => {
+      const authPayloadDto = authPayloadDtoBuilder().build();
+      const authPayload = new AuthPayload(authPayloadDto);
+
+      const newSignerAddressMock = faker.finance.ethereumAddress();
+      const mockWallet = walletBuilder().build();
+      mockWalletRepository.findOne.mockResolvedValueOnce(mockWallet);
+      mockWalletRepository.insert.mockResolvedValue({
+        identifiers: [{ id: mockWallet.id }],
+        generatedMaps: [{ id: 1 }],
+        raw: jest.fn(),
+      });
+
+      const result = await usersRepository.addWalletToUser({
+        authPayload,
+        newSignerAddress: newSignerAddressMock as `0x${string}`,
```

Comment:
When using `getAddress`, we can remove the cast here.

```suggestion
        newSignerAddress: newSignerAddressMock,
```